src/Mod/Material/MaterialEditor.py

```diff
@@ -68,8 +68,7 @@ def __init__(self, obj=None, prop=None, material=None, card_path="", category="S
         filePath = os.path.dirname(__file__) + os.sep
         self.iconPath = (filePath + "Resources" + os.sep + "icons" + os.sep)
 
-        # load the UI file from the same directory as this script
-        self.widget = FreeCADGui.PySideUic.loadUi(filePath + "Resources" + os.sep + "ui" + os.sep + "materials-editor.ui")
+        self.widget = FreeCADGui.PySideUic.loadUi(":/ui/materials-editor.ui")
```

Comment:
This was the missing piece. Well done!

Reply:
Do you mean, instead of removing the line:

```
INSTALL(FILES ${Material_Ui_Files} DESTINATION "${CMAKE_BINARY_DIR}/Mod/Material/Resources/ui")
```
to change it to

 ```
INSTALL(FILES ${Material_Ui_Files} DESTINATION "${CMAKE_INSTALL_DATADIR}/Mod/Material/Resources/ui")
```

and keeping other lines as-is? Or somethink else? If I did not get it right, pls can you post a diff or create (draft) MR of what changes are you proposing for clarity?

Reply:
@mr-miky did any of the related 3 MRs do what are you suggesting? #16069 #15565 #15714 ?

Reply:
Sorry I said it wrong.

```
@@ -169,9 +169,9 @@
 
 fc_copy_sources(MatGui "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_DATADIR}/Mod/Material" ${MatGuiIcon_SVG})
 fc_copy_sources(MatGui "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_DATADIR}/Mod/Material" ${MatGuiImages})
-fc_copy_sources(MatGui "${CMAKE_BINARY_DIR}/Mod/Material" ${Material_Ui_Files})
+fc_copy_sources(MatGui "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_DATADIR}/Mod/Material" ${Material_Ui_Files})
 
 INSTALL(TARGETS MatGui DESTINATION ${CMAKE_INSTALL_LIBDIR})
 INSTALL(FILES ${MatGuiIcon_SVG} DESTINATION "${CMAKE_INSTALL_DATADIR}/Mod/Material/Resources/icons")
 INSTALL(FILES ${MatGuiImages} DESTINATION "${CMAKE_INSTALL_DATADIR}/Mod/Material/Resources/images")
-INSTALL(FILES ${Material_Ui_Files} DESTINATION "${CMAKE_BINARY_DIR}/Mod/Material/Resources/ui")
+INSTALL(FILES ${Material_Ui_Files} DESTINATION "${CMAKE_INSTALL_DATADIR}/Mod/Material/Resources/ui")

```
[patch.txt](https://github.com/user-attachments/files/16789386/patch.txt)

Reply:
The `MaterialEditor.py` is IMHO separate problem, so I would prefer to have separate patch. The same for the other resources.

I still prefer the fix suggested in this MR, because it simplifies the code, does the "`CMAKE_INSTALL_DATADIR`" version have any benefit not mentioned in this duscussion?

Does the patch in this MR fixes the issue for you or does the issue persist?

Reply:
For me, all the problems have been solved.

Reply:
@davesrocketshop Just check I did not miss something, does this need any work, update or anythink?

Reply:
> @davesrocketshop Just check I did not miss something, does this need any work, update or anythink?

I want to test this before merging,,, I hope to do so today. I'll keep you posted